src/server/game/Spells/Spell.cpp

```diff
@@ -5122,7 +5122,7 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
 
         if (m_caster->ToUnit() && !m_caster->ToUnit()->GetSpellHistory()->IsReady(m_spellInfo, m_castItemEntry, IsIgnoringCooldowns()))
         {
-            if (m_triggeredByAuraSpell)
+            if (m_triggeredByAuraSpell || m_spellInfo->HasAttribute(SPELL_ATTR0_DISABLED_WHILE_ACTIVE))
```

Comment:
Use IsCooldownStartedOnEvent() instead of direct attribute check (spell category matters here too and that function checks it)